### PR TITLE
feat: rework autobrake keybindings

### LINKED ISSUE #2852: [REQUEST] TCA Addon Auto brake doesn’t work
**Is your feature request related to a problem? Please describe.**

**Describe the feature you'd like to see implemented**
the ThrustMaster Airbus Throttle  Addon  Auto break is not working with the FBW mod,it works with the default a320.

**References**

**Additional context**

Discord username):Azorean77

### PR DESCRIPTION
**Pull request by Saschl**
Fixes #2852

## Summary of Changes

Just a small enhancement of the update logic to repect the state of the "switch" that is being change by the keybindings.

## Additional context

Discord username (if different from GitHub): saschl#9432

## Testing instructions

Should be a quick test.

1. Add bindings to a device of your choice (autobrake min, medium and max)
2. Check that the bindings activate the corresponding autobrake level
3. Check that autobrake max can be engaged on ground
4. Check that autobrake max cannot be engaged in flight
5. Check that autobrake max does not disarm on ground
6. Perform a rejected takeoff above 70 knots and ensure the max autobrake engages and the decel light illuminates
7. Perform a landing and verify the plane is braking and the decel light illuminates

Note: currenty I see no binding to deactivate the autobrake.. if anyone got the TCA addon and could try if it deactivates through some  mystery that would be cool. Otherwise that's all for now :)

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Review by Benjozork — APPROVED** [review]
LGTM

**Comment by FlightSimmer7700** [comment]
Category: QA Tester
Name: FlightSimmer7700#3468
Date of testing: 12/1/21
Version of the sim: 1.12.13.0
PR Tested: #2854 
Object of test: AutoBrakes

Testing technique: Loaded in on the runway
Steps you've made to test the Pull Request:
1. Loaded in and hit the keyboard shortcut

BUGS / ISSUES : 

2.When pressing the keybinding it takes a second for it to go through and set it. More noticeable when moving from one to another like from LO to MED
3. took off the autobrake setting manually as the same keybind does not remove the setting set (may want to improve this) 
4.Tried to arm the autobrakes manually by pressing it but it goes back to no autobrake straight after (see video attached below)

https://user-images.githubusercontent.com/70103461/104345767-9dbc8d80-54f6-11eb-907e-0349902176f3.mp4
The issue above does not occur when loading into an approach (might be because autobreaks LO are already selected deafult)
5. When MAX auto breaks have been armed and you taxi, when you stop with the breaks the auto breaks disarm (Happens with all LO, MED and MAX){BUG}
6. After takeoff Auto breaks max stays armed and doesn't automatically disarm
7. Was on approach and set Auto breaks MED using the keybind, then I pressed the keybind for MAX and it didn't go to max stayed on MED but after landing the auto breaks went to MAX most likely because it was selected but wasn't showing it

Screenshot (438)

Flight notes (If required):

Issues (If Any): 
Overall Rating: To be improved
Conclusions of the testing: PR has a lot of bugs and issues that need to be fixed

**Comment by Saschl** [comment]
Oh my, thanks for the feedback! Yeah I might have missed that the actual "button" stays in it's position even though it's not shown which then causes some of the points at least. I thought I event checked that the behaviour in 3 and 4 does not happen in my tests but I will check again maybe my memory is wrong.
Anyway I believe further tests can wait until I reproduced the steps. Not sure if I can resolve everything but let's see

**Comment by stescouse** [comment]
Category: QA Tester
Name: Discord User stescouse
#7271
Date of testing: 15/1/2021
Version of the sim: 1.12.13.0
PR Tested: #2854
Object of test: Autobrakes

Category: QA Tester
Name: Discord User stescouse
#7271
Date of testing: 15/1/2021
Version of the sim: 1.12.13.0
PR Tested: #2854
Object of test: Autobrakes

Testing technique: Loaded Cold and Dark at gate 2 tests one with FSUICP and one without.
Steps you've made to test the Pull Request:
1. Assigned keyboard shortcuts to Low, Medium and Max Autobreaks.
2. Tested each keybind only Max autobrake worked while stationary, neither Low or Medium would engage.  
3. Whilst on taxi to runway for takeoff Max autobrake would not engage using keybind while moving.  While stationary at the gate or at runway Max autobrake engaged using keybind.
4. Whilst in air, Low and Medium Autobrake engaged using keybinds. Max Autobrake correctly would not engage in air.
5. Max Autobrake disengaged by itself on taxi before reaching runway, using keybind it engaged whilst stationary at runway. (using FSUICP)

Flight notes (If required): Some issues with FSUICP none without.

Issues (If Any): Max autobrake disengaged whilst moving towards runway on taxi using FSUICP 
Overall Rating: Room for improvement
Conclusions of the testing: As stated by other testers keybinds do not disengage autobrakes.  Autobrakes do not engage using keybinds whilst aircraft is on taxi or moving.

**Comment by Benjozork** [comment]
Is this still being worked on?

**Comment by Saschl** [comment]
Currently not, sorry. I'm still lacking the TCA addons and it doesn't behave the same with the keyboard I'm afraid..

**Comment by Benjozork** [comment]
> 
> 
> Happy to help our where I can to get this issue resolved do we need more testing?

@Saschl ?

**Comment by ErikNL80** [comment]
Hi all, I also have the TCA quadrant and add-on and more than happy to help... Stock Airbus is now fully compatible for disarm, low, medium and max, hope fbw is soon.

**Comment by ErikNL80** [comment]
Next to that, a lot has changed since January in the sim...

**Comment by tracernz** [comment]
> 
> 
> Hi all, I also have the TCA quadrant and add-on and more than happy to help... Stock Airbus is now fully compatible for disarm, low, medium and max, hope fbw is soon.

What do you have each of those bound to in the sim?

**Comment by Saschl** [comment]
Thanks for the hint, I can try to replicate the behaviour I should have some time now.. but as already said, a screenshot from the bindings would also help, thanks!

**Comment by ErikNL80** [comment]
Below is a screenshot of the current bindings. This is part of the default profile that comes with the TCA Quadrant/Add-on and is confirmed working in vanilla A320

Naamloos

**Comment by Saschl** [comment]
I have no clue why it decided to close the PR now, sorry. @ErikNL80 I have made some progress, it does work for me now with the keyboard and a rotary switch (still unable to get the addons currently :/). If you're a bit into testing I could leave a link here with a version that includes the changes. It still needs some work but I think the functionality is there now

**Comment by ErikNL80** [comment]
@Saschl Sure, be happy to test!

**Comment by Saschl** [comment]
Cool! https://github.com/flybywiresim/a32nx/suites/2585233660/artifacts/56422463 Extract this to your community folder (be sure to remove existing installations first). Also there's a zip in a zip file. Extract both until you can see the "flybywire-aircraft-a320-neo" folder, that one goes into community then.  You can revert anytime by deleting the folder and using the installer again.

**Comment by ninoz** [comment]
Can confirm it works  perfectly with the TCA addon :)

Low/Med/Max can be selected on the ground and Low/Med can only be selected in the air as expected :)

Nice work!

**Comment by ErikNL80** [comment]
@Saschl I do not filled a bug template, but tested if behaviour is the same as with the vanilla A320neo. And it works great! Just as @ninoz reports.

**Comment by Saschl** [comment]
@ninoz @ErikNL80 thank you it’s fine like this, and sorry for all the delay! Disarming autobrake works as well? 
I need to clean it up a bit still but then we should be fine.

**Comment by ninoz** [comment]
Yes disarming works too :)

**Comment by ErikNL80** [comment]
@Benjozork What is the next step to take?

**Comment by Saschl** [comment]
Next step is on me because the code is messy :D Then I'll let it review again. Until then back to draft (but it won't be long I just can't do this today anymore)

**Comment by ErikNL80** [comment]
No problem! Good progress is made ;) Thanks!

**Comment by ErikNL80** [comment]
It seems that this PR is now ready for reviewing, but the label still indicates 'Waiting For Response'. Can the label be updated so everyone knows progress is made?

**Comment by Benjozork** [comment]
@wpine215 fyi

**Review by Benjozork — APPROVED** [review]

**Comment by nyhmbo** [comment]
Category: Contributor
Name: Nyhmbo#9999
Date of testing: 07/05/2021
Version of the sim: 1.15.10.0
PR Tested: #2854 
Tier of testing: 1
Changes to observe:
autobrake keybinds

Testing technique: Test

Steps you've made to test the Pull Request:
1. load c/d and configure the plane
2. test autobrake keybinds on ground using TCA addon and keyboard (LO, MED and MAX)
3. check automatic autobrake disarming after gear is retracted
4. check if autobrake LO and MED can be selected while in the air
4. check that autobrake max is not possible while in the air
5. check autobrake activation + disarm on landing (LO and MED)

Flight notes (When done a full flight): none

Media: Insert screenshots, recordings, or anything relevant to the review here.

Issues (If Any): none

Overall Rating: Good
Conclusions of the testing: works like it should, even with the TCA. the only thing that I noticed is that while using the TCA rotary switch is that it arms again after automatically disarming on landing if you don't switch to the disarm position manually, don't see that as a big issue tho.

File: src/behavior/src/A32NX_Interior_Misc.xml
Comment (Saschl) [reply src/behavior/src/A32NX_Interior_Misc.xml:170]: Heh, so the condition for the update code is a bit misleading. As this template is used by all the three buttons, it would cause the update to run three times, this condition prevents that as the code itself still affects all levels potentially. 

The condition you mentioned is only important when spawning on the runway (makes sure the initial value does not get overriden) but the assignment is useless. I removed it.

**Comment by FlightSimmer7700** [comment]
Category: QA Tester
Name: FlightSimmer7700#3468
Date of testing: 8/5/21
Version of the sim: 1.15.10.0
PR Tested: #2854 
Tier of testing: 1
Changes to observe:
Aurobrake keybindings

Testing technique: Loaded in at different points and tried
Steps you've made to test the Pull Request:
1. Loaded at gate and tried
2. Loaded on Runway and tried
3. Loaded on approach and tried

Flight notes (When done a full flight):

Media: Insert screenshots, recordings, or anything relevant to the review here.
image


Issues (If Any): Just like on the last report from 4 months ago when on approach and selecting low or medium it works fine but when selecting one of those and then max using the keybind it won't show up as "ON" but after landing auto brake "ON" light will move to the max auto brakes causing the plane to brake at maximum even though it doesn't show up on approach

Steps to achieve the issue: Load up on approach and select Low or medium using keybind or button and then select Max using the keybind and on landing, you should notice it going to max auto brake

Overall Rating: To be improved
Conclusions of the testing: Overall improved in other areas but still has the old issue

**Comment by FlightSimmer7700** [comment]
> Category: QA Tester
> Name: FlightSimmer7700#3468
> Date of testing: 8/5/21
> Version of the sim: 1.15.10.0
> PR Tested: #2854
> Tier of testing: 1
> Changes to observe:
> Aurobrake keybindings
> 
> Testing technique: Loaded in at different points and tried
> Steps you've made to test the Pull Request:
> 
> 1. Loaded at gate and tried
> 2. Loaded on Runway and tried
> 3. Loaded on approach and tried
> 
> Flight notes (When done a full flight):
> 
> Media: Insert screenshots, recordings, or anything relevant to the review here.
> image
> 
> Issues (If Any): Just like on the last report from 4 months ago when on approach and selecting low or medium it works fine but when selecting one of those and then max using the keybind it won't show up as "ON" but after landing auto brake "ON" light will move to the max auto brakes causing the plane to brake at maximum even though it doesn't show up on approach
> 
> Steps to achieve the issue: Load up on approach and select Low or medium using keybind or button and then select Max using the keybind and on landing, you should notice it going to max auto brake
> 
> Overall Rating: To be improved
> Conclusions of the testing: Overall improved in other areas but still has the old issue

Checked after recent changes and the auto brakes now work as intended after touchdown

**Pull request merged**
